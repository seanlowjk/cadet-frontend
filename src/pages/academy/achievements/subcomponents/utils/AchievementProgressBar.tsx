import React from 'react';
import { ProgressBar } from '@blueprintjs/core';

type AchievementProgressBarProps = {
  completionProgress: number;
  completionGoal: number;
  shouldAnimate: boolean;
};

function AchievementProgressBar(props: AchievementProgressBarProps) {
  const { completionProgress, completionGoal, shouldAnimate } = props;

  const progressInDecimal: number = Math.min(completionProgress / completionGoal, 1);

  return (
    <div>
      <ProgressBar
        intent={progressInDecimal === 1 ? 'success' : undefined}
        value={progressInDecimal}
        animate={shouldAnimate}
      />
    </div>
  );
}

export default AchievementProgressBar;
