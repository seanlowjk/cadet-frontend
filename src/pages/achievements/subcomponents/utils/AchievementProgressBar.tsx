import React from 'react';
import { ProgressBar } from '@blueprintjs/core';

type AchievementProgressBarProps = {
  progressFrac: number;
  shouldAnimate: boolean;
};

function AchievementProgressBar(props: AchievementProgressBarProps) {
  const { progressFrac, shouldAnimate } = props;

  return (
    <div>
      <ProgressBar
        className="progress"
        intent={progressFrac === 1 ? 'success' : undefined}
        value={progressFrac}
        animate={shouldAnimate}
      />
    </div>
  );
}

export default AchievementProgressBar;
