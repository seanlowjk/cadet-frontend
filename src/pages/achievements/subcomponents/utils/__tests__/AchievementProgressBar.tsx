import { mount } from 'enzyme';
import React from 'react';

import AchievementProgressBar from '../AchievementProgressBar';

const incompleteProgressProps = {
  progressFrac: 0,
  shouldAnimate: false
};

const completeProgressProps = {
  progressFrac: 1,
  shouldAnimate: false
};

test('AchievementProgressBar component renders correctly', () => {
  const incompleteProgress = <AchievementProgressBar {...incompleteProgressProps} />;
  const tree = mount(incompleteProgress);
  expect(tree.debug()).toMatchSnapshot();

  const completeProgress = <AchievementProgressBar {...completeProgressProps} />;
  const tree_1 = mount(completeProgress);
  expect(tree_1.debug()).toMatchSnapshot();
});
