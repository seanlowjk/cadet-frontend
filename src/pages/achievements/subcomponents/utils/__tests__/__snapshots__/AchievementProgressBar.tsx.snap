// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AchievementProgressBar component renders correctly 1`] = `
"<AchievementProgressBar progressFrac={0} shouldAnimate={false}>
  <div>
    <Blueprint3.ProgressBar className=\\"progress\\" intent={[undefined]} value={0} animate={false}>
      <div className=\\"bp3-progress-bar bp3-no-animation progress\\">
        <div className=\\"bp3-progress-meter\\" style={{...}} />
      </div>
    </Blueprint3.ProgressBar>
  </div>
</AchievementProgressBar>"
`;

exports[`AchievementProgressBar component renders correctly 2`] = `
"<AchievementProgressBar progressFrac={1} shouldAnimate={false}>
  <div>
    <Blueprint3.ProgressBar className=\\"progress\\" intent=\\"success\\" value={1} animate={false}>
      <div className=\\"bp3-progress-bar bp3-intent-success bp3-no-animation progress\\">
        <div className=\\"bp3-progress-meter\\" style={{...}} />
      </div>
    </Blueprint3.ProgressBar>
  </div>
</AchievementProgressBar>"
`;
